README.md

```diff
@@ -1,178 +1,317 @@
+Before running specs, you'll need to install [Ruby][] and [Bundler][]. The spec
+runner is written in Ruby, so it's necessary no matter which implementation
+you're testing. Then, from the root of this repo, run `bundle install`.
 
-Run tests against Ruby Sass with the `sass-spec.rb` file in the root directory.
+[Ruby]: https://www.ruby-lang.org/en/
+[Bundler]: http://bundler.io/
 
-    $ git clone https://github.com/sass/sass-spec.git
-    $ cd sass-spec
-    $ bundle install
-    $ bundle exec sass-spec.rb
+From there, it depends which implementation you're testing:
 
-To run tests against Ruby Sass in development, edit the Gemfile and add:
+### Dart Sass
+
+To run specs against [Dart Sass][], the reference implementation of Sass that's
+used for [the `sass` package][] on npm, you'll first need to install [Dart][].
+Then run:
+
+[Dart Sass]: https://sass-lang.com/dart-sass
+[the `sass` package]: https://npmjs.com/package/sass
+[Dart]: https://www.dartlang.org/
+
+```sh
+# If you already have a clone of the Dart Sass repo, you can use that instead.
+git clone https://github.com/sass/dart-sass
+(cd dart-sass; pub get)
+export DART_SASS_PATH=`pwd`/dart-sass
+
+bundle exec ./sass-spec.rb --dart $DART_SASS_PATH
+```
+
+### LibSass
+
+To run specs against [LibSass][], the C++ Sass implementation that's used for
+[Node Sass][] and other languages' Sass wrappers, you'll need to be able to
+[build LibSass][]. Once you have all the build dependencies:
+
+[LibSass]: https://sass-lang.com/libsass
+[Node Sass]: https://npmjs.com/package/node-sass
+[build LibSass]: https://github.com/sass/libsass/blob/master/docs/build.md
+
+```sh
+# If you already have a clone of the LibSass repo, you can use that instead.
+git clone https://github.com/sass/libsass
+(cd libsass; ./script/bootstrap; make sassc)
+export SASSC_PATH=`pwd`/libsass/sassc/bin/sassc
+
+bundle exec ./sass-spec.rb --impl libsass -c $SASSC_PATH
+```
+
+### Ruby Sass
+
+Unlike other implementations, Ruby Sass is installed via Bundler and used by
+default. All you need to do to run specs against it is:
+
+```sh
+bundle exec ./sass-spec.rb
+```
+
+If you want to run specs against your own clone of Ruby Sass, edit the `Gemfile`
+and replace the `gem "sass"` line with:
 
 ```ruby
-gem 'sass', :path => "/path/to/sass/directory"
+gem "sass", :path => "path/to/ruby/sass"
 ```
 
-Then run `bundle install` and `bundle exec sass-spec.rb` will run
-against your development version of ruby sass.
+## Spec Structure
+
+Each spec is defined by a directory with an `input.scss` or `input.sass` file
+and either:
+
+* An `output.css` file, in which case the spec asserts that the Sass
+  implementation compiles the input to the output. These specs are known as
+  "success specs".
+* An `error` file, in which case the spec asserts that the Sass implementation
+  prints the error message to standard error and exits with a non-zero status
+  code when it compiles the input. These specs are known as "error specs".
+
```

Comment:
Maybe add a short callout with link to the "implementation-specific-expectations" section here.

Reply:
Done.